src/main/java/Duke.java

```diff
@@ -1,10 +1,318 @@
+            System.out.println("File i/o error.");
+            return null;
+        }
+    }
+
+    // ArrayLists are passed by reference
+    public static void insertPreviousItems(File f, ArrayList<Task> tasks) {
+        try {
+            Scanner fileScanner = new Scanner(f);
+        
+            while (fileScanner.hasNext()) {
+                String lineInput = fileScanner.nextLine();
+                String[] words = lineInput.split("\\|", 4);
+
+                switch (words[0]) {
+                case "T":
+                    Todo newTodo = new Todo(words[2]);
+                    if (words[1].equals("Y")) {
+                        newTodo.markDone();
+                    }
+                    tasks.add(newTodo);
+                    break;
+
+                case "E":
+                    Event newEvent = new Event(words[2], words[3]);
+                    if (words[1].equals("Y")) {
+                        newEvent.markDone();
+                    }
+                    tasks.add(newEvent);
+                    break;
+
+                case "D":
+                    Deadline newDeadline = new Deadline(words[2], words[3]);
+                    if (words[1].equals("Y")) {
+                        newDeadline.markDone();
+                    }
+                    tasks.add(newDeadline);
+                    break;
+
+                default:
+                    break;
+                }
+            }
+            fileScanner.close();
+
+        } catch (FileNotFoundException e) {
+            System.out.println("error accessing file");
+        }    
+    }
+
+    public static void updateFile(ArrayList<Task> tasks) { 
+        try {
+            File f = new File("data/stored.txt");
+            if (f.exists()) {
+                f.delete();
+            }
+            f.createNewFile();
+
+            FileWriter fw = new FileWriter(f);
+            for (int i = 0; i < tasks.size(); i++) {
+                fw.write(tasks.get(i).createFileString() + "\n");
+            }
+
+            fw.close();
+        } catch (Exception e) {
+            System.out.println("File i/o error");
+        }
+    }
+
+    public static void listTasks(ArrayList<Task> tasks) {
+        String line = "\t------------------------\n";
+        System.out.println(line);
+        for (int i = 0; i < tasks.size(); i++) {
+            int j = i + 1;
+            System.out.println("\t"+ j + ". " + tasks.get(i) + "\n");
+        }
+        System.out.println(line);
+    }
+
+    public static void printStatus(Task latest, int total, boolean isAdd){
+        String verb = "added";
+        if (!isAdd) {
+            verb = "removed";
+        }
+
+        String line = "\t------------------------\n";
+        System.out.println(line + "\tGot it. I've " + verb + " this task:\n\t" + latest);
+        System.out.println("\n\tNow you have " + total + " task(s) in the list.\n" + line);
+    }
+
+    public static void processUserInput() {
```

Comment:
This method may be too long.

https://nus-cs2113-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods